lib/fedora_migrate/datastream_verification.rb

```diff
@@ -4,29 +4,18 @@ module FedoraMigrate::DatastreamVerification
 
   def valid? datastream=nil
     @datastream = datastream || @source
-    check = has_no_checksum? || has_matching_checksums? || content_equivalent?
+    check = has_matching_checksums? || has_matching_nokogiri_checksums?
     FedoraMigrate::Logger.warn "#{@datastream.pid} datastream #{@datastream.dsid} validation failed" unless check
     check
   end
 
   def has_matching_checksums?
-    datastream.checksum == target_checksum
+    datastream.checksum == target_checksum || checksum(datastream.content) == target_checksum
```

Comment:
Like I said, :+1:, though I might like to see this method look like:

```ruby
def has_matching_checksums?
  checksum(datastream) == target_checksum
end
```

```ruby
def checksum content
  content.respond_to?(:checksum) ? content.checksum : Digest::SHA1.hexdigest(content)
end
```

But it's :peach:-y as is!